compact_str/src/repr2/heap.rs

```diff
@@ -16,6 +16,19 @@ const MIN_HEAP_SIZE: usize = MAX_SIZE + mem::size_of::<usize>();
 const UNKNOWN: usize = 0;
 pub type StrBuffer = [u8; UNKNOWN];
 
+/// [`HeapBuffer`] grows at an amortized rates of 1.5x
+///
+/// Note: this is different than [`std::string::String`], which grows at a rate of 2x. It's debated
+/// which is better, for now we'll stick with a rate of 1.5x
+///
+/// TODO: Handle overflows in the case of __very__ large Strings
+#[inline(always)]
+pub fn amortized_growth(cur_len: usize, additional: usize) -> usize {
+    let required = cur_len + additional;
+    let amortized = 3 * cur_len / 2;
+    core::cmp::max(amortized, required)
```

Comment:
Nit: Could use `usize::max` instead:

```suggestion
    amortized.max(required)
```